sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2727,6 +2727,12 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
       t => t.containsAnyPattern(AGGREGATE_EXPRESSION, PYTHON_UDF) && t.containsPattern(PROJECT),
       ruleId) {
       case Project(projectList, child) if containsAggregates(projectList) =>
+        if (child.isInstanceOf[PipeOperatorSelect]) {
```

Comment:
shall we move this to `CheckAnalysis`?

Reply:
Good Q, by that point `RemovePipeOperators` will have removed the `PipeOperatorSelect` so we won't be able to tell anymore, so we have to check it here.

Reply:
We can move `RemovePipeOperators` to `FinishAnalysis`

Reply:
I tried this but it was more complex than I had hoped. I ended up switching the implementation to make the `PipeOperatorSelect` a temporary expression instead, that checks its children for aggregate functions at the end of analysis. This seems to work without creating any new analyzer rule.